src/main/java/com/mongodb/starter/repositories/MongoDBPersonRepository.java

```diff
@@ -0,0 +1,137 @@
+package com.mongodb.starter.repositories;
+
+import com.mongodb.ReadConcern;
+import com.mongodb.ReadPreference;
+import com.mongodb.TransactionOptions;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.FindOneAndReplaceOptions;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.WriteModel;
+import com.mongodb.starter.dtos.AverageAgeDTO;
+import com.mongodb.starter.models.Person;
+import org.bson.BsonDocument;
+import org.bson.BsonNull;
+import org.bson.conversions.Bson;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.mongodb.client.model.Accumulators.avg;
+import static com.mongodb.client.model.Aggregates.group;
+import static com.mongodb.client.model.Aggregates.project;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.in;
+import static com.mongodb.client.model.Projections.excludeId;
+import static java.util.Arrays.asList;
+
+@Repository
+public class MongoDBPersonRepository implements PersonRepository {
+
+    private static final TransactionOptions txnOptions = TransactionOptions.builder()
+                                                                           .readPreference(ReadPreference.primary())
+                                                                           .readConcern(ReadConcern.MAJORITY)
+                                                                           .writeConcern(WriteConcern.MAJORITY)
+                                                                           .build();
+    @Autowired
+    private MongoClient client;
+    private MongoCollection<Person> personCollection;
+
+    @PostConstruct
+    void init() {
+        personCollection = client.getDatabase("test").getCollection("persons", Person.class);
```

Comment:
I noticed you included the database name in the default URI, but then didn't use it.  You could consider auto-wiring it into this class along with the client.

Reply:
I removed the database from the URI so the template does not depend on what people will set in the URI. Let's stay as generic and simple as possible.

Reply:
There is no resolve button... I don't know why.

Resolved.